keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/BaseClusterHandler.kt

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.keel.api.plugins
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceDiff
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.diff.toIndividualDiffs
+
+/**
+ * Common cluster functionality.
+ * This class contains abstracted logic where we want all clusters to make the
+ * same decision, or abstracted logic involving coordinating resources or
+ * publishing events.
+ */
+abstract class BaseClusterHandler<SPEC: ResourceSpec, RESOLVED: Any>(
+  resolvers: List<Resolver<*>>
+) : ResolvableResourceHandler<SPEC, Map<String, RESOLVED>>(resolvers) {
+
+  /**
+   * parses the desired region from a resource diff
+   */
+  abstract fun getDesiredRegion(diff: ResourceDiff<RESOLVED>): String
```

Comment:
Is that just returning the region name? Or is it some chunk of diff JSON?

Also both implementations are identical so I'm not sure why it's abstract.

Reply:
just region name. Implementations are identical, yes, but that's because the spec is structured these same for both classes. You need to have type information in order to actually get to the region

Reply:
Is that something we could unify?

Reply:
Only by making a base cluster spec, which I don't want to do right now / in this PR